### PR TITLE
Adjust memory limit for heapster's pod nanny

### PR DESCRIPTION
**Pull request by gmarek**
cc @piosz @wojtek-t @zmerlynn

### CODE REVIEW

**Comment by wojtek-t** [comment]
@roberthbailey

**Comment by wojtek-t** [comment]
LGTM from my perspective

**Comment by piosz** [comment]
More context: nanny is OOMing on 2k nodes cluster.

**Comment by roberthbailey** [comment]
Can you summarize in the PR comments what the changes are? Does it change across the board or only at the high end when running on bigger clusters?

**Comment by wojtek-t** [comment]
It is change for all sizes of the cluster - 200kB of memory request for every node.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -708,12 +708,15 @@ function start-kube-addons {
     metrics_memory="${base_metrics_memory}"
     base_eventer_memory="200Mi"
     eventer_memory="${base_eventer_memory}"
+    nanny_memory="100Mi"
```
Comment (andyzheng0831) [inline cluster/gce/gci/configure-helper.sh]: please define it with 'local'

**Comment by andyzheng0831** [comment]
The GCI config change lgtm, To be safe, I would suggest to spin up a cluster, log into master, enter /etc/manifest/addons to find the heapster yaml file, and check if the new variable is correctly evaluated.

**Comment by Q-Lee** [comment]
Whoa, I wouldn't have guessed the nanny would OOM on any realistically sized cluster.

**Comment by wojtek-t** [comment]
@roberthbailey @Q-Lee - are you fine with this PR? We really need something like that for 1.3 so it is pretty important to know your opinion asap.

**Comment by Q-Lee** [comment]
lgtm if tests pass.

**Comment by dchen1107** [comment]
@k8s-bot test this

**Comment by wojtek-t** [comment]
@gmarek - it seems that you need to rebase to make tests passing

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (gmarek) [inline cluster/gce/gci/configure-helper.sh]: Why nanny memory should be treated differently than eventer memory?

File: cluster/addons/cluster-monitoring/googleinfluxdb/heapster-controller-combined.yaml [hunk 2]
```diff
@@ -5,9 +5,12 @@
 {% set metrics_memory_per_node = 4 -%}
 {% set eventer_memory_per_node = 500 -%}
 {% set num_nodes = pillar.get('num_nodes', -1) -%}
+{% set nanny_memory = "100Mi" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/googleinfluxdb/heapster-controller-combined.yaml]: please change to 90 (as we don't want to increase resources for small clusters)

File: cluster/addons/cluster-monitoring/googleinfluxdb/heapster-controller-combined.yaml [hunk 3]
```diff
@@ -5,9 +5,12 @@
...
+{% set nanny_memory_per_node = 200 -%}
 {% if num_nodes >= 0 -%}
   {% set metrics_memory = (200 + num_nodes * metrics_memory_per_node)|string + "Mi" -%}
   {% set eventer_memory = (200 * 1024 + num_nodes * eventer_memory_per_node)|string + "Ki" -%}
+  {% set nanny_memory = (100 * 1024 + num_nodes * nanny_memory_per_node)|string + "Ki" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/googleinfluxdb/heapster-controller-combined.yaml]: and here

File: cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml [hunk 4]
```diff
@@ -5,9 +5,12 @@
 {% set metrics_memory_per_node = 4 -%}
 {% set eventer_memory_per_node = 500 -%}
 {% set num_nodes = pillar.get('num_nodes', -1) -%}
+{% set nanny_memory = "100Mi" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml]: please change to 90 (as we don't want to increase resources for small clusters)

File: cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml [hunk 5]
```diff
@@ -5,9 +5,12 @@
...
+{% set nanny_memory_per_node = 200 -%}
 {% if num_nodes >= 0 -%}
   {% set metrics_memory = (200 + num_nodes * metrics_memory_per_node)|string + "Mi" -%}
   {% set eventer_memory = (200 * 1024 + num_nodes * eventer_memory_per_node)|string + "Ki" -%}
+  {% set nanny_memory = (100 * 1024 + num_nodes * nanny_memory_per_node)|string + "Ki" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml]: and here

File: cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml [hunk 6]
```diff
@@ -2,8 +2,11 @@
 {% set metrics_memory = base_metrics_memory -%}
 {% set metrics_memory_per_node = 4 -%}
 {% set num_nodes = pillar.get('num_nodes', -1) -%}
+{% set nanny_memory = "100Mi" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml]: please change to 90 (as we don't want to increase resources for small clusters)

File: cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml [hunk 7]
```diff
@@ -2,8 +2,11 @@
...
+{% set nanny_memory_per_node = 200 -%}
 {% if num_nodes >= 0 -%}
   {% set metrics_memory = (200 + num_nodes * metrics_memory_per_node)|string + "Mi" -%}
+  {% set nanny_memory = (100 * 1024 + num_nodes * nanny_memory_per_node)|string + "Ki" -%}
```
Comment (wojtek-t) [inline cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml]: and here

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (wojtek-t) [inline cluster/gce/gci/configure-helper.sh]: I think it shouldn't.

please change to 90 (as we don't want to increase resources for small clusters)

File: cluster/gce/gci/configure-helper.sh [hunk 8]
```diff
@@ -725,12 +725,15 @@ function start-kube-addons {
...
     local -r metrics_memory_per_node="4"
     local -r eventer_memory_per_node="500"
+    local -r nanny_memory_per_node="200"
     if [[ -n "${NUM_NODES:-}" && "${NUM_NODES}" -ge 1 ]]; then
       num_kube_nodes="$((${NUM_NODES}-1))"
       metrics_memory="$((${num_kube_nodes} * ${metrics_memory_per_node} + 200))Mi"
       eventer_memory="$((${num_kube_nodes} * ${eventer_memory_per_node} + 200 * 1024))Ki"
+      nanny_memory="$((${num_kube_nodes} * ${nanny_memory_per_node} + 100 * 1024))Ki"
```
Comment (wojtek-t) [inline cluster/gce/gci/configure-helper.sh]: s/100/90/

File: cluster/gce/trusty/configure-helper.sh [hunk 9]
```diff
@@ -717,12 +717,15 @@ start_kube_addons() {
     metrics_memory="${base_metrics_memory}"
     base_eventer_memory="200Mi"
     eventer_memory="${base_eventer_memory}"
+    nanny_memory="100Mi"
```
Comment (wojtek-t) [inline cluster/gce/trusty/configure-helper.sh]: please change to 90 (as we don't want to increase resources for small clusters)

File: cluster/gce/trusty/configure-helper.sh [hunk 10]
```diff
@@ -717,12 +717,15 @@ start_kube_addons() {
...
     readonly metrics_memory_per_node="4"
     readonly eventer_memory_per_node="500"
+    readonly nanny_memory_per_node="200"
     if [ -n "${NUM_NODES:-}" ] && [ "${NUM_NODES}" -ge 1 ]; then
       num_kube_nodes="$((${NUM_NODES}-1))"
       metrics_memory="$((${num_kube_nodes} * ${metrics_memory_per_node} + 200))Mi"
       eventer_memory="$((${num_kube_nodes} * ${eventer_memory_per_node} + 200 * 1024))Ki"
+      nanny_memory="$((${num_kube_nodes} * ${nanny_memory_per_node} + 100 * 1024))Ki"
```
Comment (wojtek-t) [inline cluster/gce/trusty/configure-helper.sh]: s/100/90/

**Comment by wojtek-t** [comment]
LGTM - thanks

**Comment by wojtek-t** [comment]
Just to clarify - the after some tuning, the net effect is that for small clusters, we are not increasing memory (for clusters with <=50 nodes) [it's even slightly decreased]. Thus I think the change is not controversial.
@roberthbailey @Q-Lee - FYI ^^

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please, issue: #IGNORE (no test results)

**Comment by zmerlynn** [comment]
Nudging this up - we need to get heapster testing on large clusters finished today, and it's long.

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please, issue: #27344 27344

**Comment by Q-Lee** [comment]
lgtm
